sql queries.txt

```diff
@@ -0,0 +1,1985 @@
+    -> FROM SALESPEOPLE
+    -> JOIN ORDERS ON SALESPEOPLE.SNUM = ORDERS.SNUM
+    -> GROUP BY SALESPEOPLE.SNAME, ORDERS.ODATE;
++---------+------------+---------------+
+| SNAME   | ODATE      | largest_order |
++---------+------------+---------------+
+| AxelRod | 1990-10-04 |       1713.23 |
+| Motika  | 1990-10-03 |       1900.10 |
+| Peel    | 1990-10-03 |        767.19 |
+| Peel    | 1990-10-05 |       4723.00 |
+| Peel    | 1990-10-06 |       9891.88 |
+| Rifkin  | 1990-10-03 |       1098.16 |
+| Serres  | 1990-10-03 |       5160.45 |
+| Serres  | 1990-10-04 |         75.75 |
+| Serres  | 1990-10-06 |       1309.95 |
++---------+------------+---------------+
+9 rows in set (0.02 sec)
+
+
+
+
+
+4) Arrange the Order table by descending customer number. 
+
+mysql> SELECT * FROM CUSTOMERS
+    -> ORDER BY CNUM DESC;
++------+----------+---------+--------+------+
+| CNUM | CNAME    | CITY    | RATING | SNUM |
++------+----------+---------+--------+------+
+| 2008 | Cisneros | SanJose |    300 | 1007 |
+| 2007 | Pereira  | Rome    |    100 | 1004 |
+| 2006 | Clemens  | London  |    100 | 1001 |
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2003 | Liu      | SanJose |    200 | 1002 |
+| 2002 | Giovanni | Rome    |    200 | 1003 |
+| 2001 | Hoffman  | London  |    100 | 1001 |
++------+----------+---------+--------+------+
+7 rows in set (0.00 sec)
+
+
+
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> SELECT SALESPEOPLE.SNAME, SALESPEOPLE.SNUM FROM SALESPEOPLE,ORDERS
+    -> WHERE SALESPEOPLE.SNUM=ORDERS.SNUM
+    -> GROUP BY SNAME;
++---------+------+
+| SNAME   | SNUM |
++---------+------+
+| AxelRod | 1003 |
+| Motika  | 1004 |
+| Peel    | 1001 |
+| Rifkin  | 1007 |
+| Serres  | 1002 |
++---------+------+
+5 rows in set (0.00 sec)
+
+
+
+
+
+6) List names of all customers matched with the salespeople serving them. 
+
+mysql> SELECT CUSTOMERS.CNAME, SALESPEOPLE.SNAME AS salespeople_serving FROM SALESPEOPLE
+    -> JOIN CUSTOMERS ON SALESPEOPLE.SNUM=CUSTOMERS.SNUM;
++----------+---------------------+
+| CNAME    | salespeople_serving |
++----------+---------------------+
+| Hoffman  | Peel                |
+| Giovanni | AxelRod             |
+| Liu      | Serres              |
+| Grass    | Serres              |
+| Clemens  | Peel                |
+| Cisneros | Rifkin              |
+| Pereira  | Motika              |
++----------+---------------------+
+
+
+
+
+
+7) Find the names and numbers of all salespeople who have more than one customer. 
+
+mysql> SELECT SALESPEOPLE.SNAME, SALESPEOPLE.SNUM FROM SALESPEOPLE
+    -> JOIN CUSTOMERS ON SALESPEOPLE.SNUM=CUSTOMERS.SNUM
+    -> GROUP BY SNUM HAVING COUNT(SALESPEOPLE.SNUM)>1;
++--------+------+
+| SNAME  | SNUM |
++--------+------+
+| Peel   | 1001 |
+| Serres | 1002 |
++--------+------+
+2 rows in set (0.00 sec)
+
+
+
+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
Wrong 
Try to think about this 
if there is salespeople who is not associated with any order 
Expected
In that case the order count of salespeople  shows as 0.

Reply:
you are right sir, I should have used left join here.
I have updated the query.